examples/boost/CMakeLists.txt

```diff
@@ -4,19 +4,34 @@ project(CPMExampleBoost)
 
 # ---- Create binary ----
 
-add_executable(CPMExampleBoost main.cpp)
+add_executable(${PROJECT_NAME} main.cpp)
 target_compile_features(CPMExampleBoost PRIVATE cxx_std_17)
 
 # ---- Dependencies ----
 
 include(../../cmake/CPM.cmake)
 
 CPMAddPackage(
-  NAME Boost
-  VERSION 1.84.0
-  URL https://github.com/boostorg/boost/releases/download/boost-1.84.0/boost-1.84.0.tar.xz
-  URL_HASH SHA256=2e64e5d79a738d0fa6fb546c6e5c2bd28f88d268a2a080546f74e5ff98f29d0e
-  OPTIONS "BOOST_ENABLE_CMAKE ON" "BOOST_INCLUDE_LIBRARIES container\\\;asio" # Note the escapes!
+  NAME AddBoost.CMake
+  GIT_TAG main
+  GITHUB_REPOSITORY Arniiiii/AddBoost.cmake
```

Comment:
This example lacks security and a specific version of boost. When developing a commercial application, there's often a _need_ to call out a specific revision of boost. 

These lines are neither equivalent nor better than the removed lines for my purposes.

Reply:
Security? Not sure. I admit the problem with specific version. I'll try to solve it

Reply:
I've thought a little bit about version handling. If to think about AddBoost.cmake not as wrapper for Boost but just another cmake script, In the end it's still `find_package(Boost COMPONENTS ${BOOST_NOT_HEADER_ONLY_COMPONENTS}) ` or CPMAddPackage(NAME Boost VERSION ${TRY_BOOST_VERSION} )` .

If to think about specific revision of boost there's two options: 
1. Use specific boost version for which there's tag and patch it
2. If to think about specifying SOURCE_DIR for CPM to get a separate boost. 
It's easy to add the ability to do the second one  to the AddBoost.cmake